test/Microsoft.TemplateEngine.Edge.UnitTests/GlobalSettingsTests.cs

```diff
@@ -52,7 +52,15 @@ public async Task TestFilwatcher()
             using var globalSettings1 = new GlobalSettings(envSettings, settingsFile);
             using var globalSettings2 = new GlobalSettings(envSettings, settingsFile);
             var taskSource = new TaskCompletionSource<TemplatePackageData>();
-            globalSettings2.SettingsChanged += async () => taskSource.TrySetResult((await globalSettings2.GetInstalledTemplatePackagesAsync(default).ConfigureAwait(false)).Single());
+            globalSettings2.SettingsChanged += async () => {
+                var packages = await globalSettings2.GetInstalledTemplatePackagesAsync(default).ConfigureAwait(false);
```

Comment:
is it an issue for template engine itself? should we do similar in template engine code?

Reply:
As far as I understand, it will trigger `TemplatePackagesChanged` event multiple times and last time will be correct. There isn't much better we can do in engine code I think...

Reply:
Should we add the bug to some repo about watcher? :)

Reply:
I think not, I think problem is just that on Linux filewatcher is a bit too fast :)